adapter/stdioLogger/stdioLogger.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package stdLogger provides an implementation of the mixer logger aspect
+// that writes logs (serialized as JSON) to a standard expectedStream (stdout | stderr).
+package stdLogger
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"istio.io/mixer/adapter/stdioLogger/config"
+	"istio.io/mixer/pkg/aspect"
+	"istio.io/mixer/pkg/aspect/logger"
+	"istio.io/mixer/pkg/registry"
+
+	me "github.com/hashicorp/go-multierror"
+)
+
+type (
+	adapter    struct{}
+	aspectImpl struct {
+		logStream       io.Writer
+		logName         string
+		payloadAttr     string
+		payloadFormat   payloadFormat
+		severityAttr    string
+		timestampAttr   string
+		timestampFormat string
+		timeFn          func() time.Time
+	}
+
+	payloadFormat int
+
+	logEntry struct {
+		Name          string                 `json:"logName,omitempty"`
+		Timestamp     time.Time              `json:"timestamp,omitempty"`
+		Labels        map[string]string      `json:"labels,omitempty"`
+		Severity      string                 `json:"severity,omitempty"`
+		TextPayload   string                 `json:"textPayload,omitempty"`
+		StructPayload map[string]interface{} `json:"structPayload,omitempty"`
+	}
+)
+
+const (
+	textFmt   payloadFormat = iota
+	structFmt payloadFormat = iota
+)
+
+// Register adds the stdLogger adapter to the list of logger.Aspects known to
+// mixer.
+func Register(r registry.Registrar) error { return r.RegisterLogger(&adapter{}) }
+
+func (a *adapter) Name() string { return "istio/stdLogger" }
+func (a *adapter) Description() string {
+	return "Writes log entries to a standard I/O stream with JSON encoding"
+}
+func (a *adapter) DefaultConfig() proto.Message { return &config.Params{} }
+func (a *adapter) Close() error                 { return nil }
+func (a *adapter) ValidateConfig(cfg proto.Message) (ce *aspect.ConfigErrors) {
+	if _, ok := cfg.(*config.Params); ok {
+		return nil
+	}
+
+	return ce.Append("", fmt.Errorf("unexpected type for proto: %T", cfg))
+}
+func (a *adapter) NewAspect(env aspect.Env, cfg proto.Message) (logger.Aspect, error) {
+	c := cfg.(*config.Params)
+
+	w := os.Stderr
+	if c.GetLogStream() == config.Params_STDOUT {
```

Comment:
We can assume that c != nil. If a config doesn't exist, the environment will pass DefaultConfig instead. So we can access proto message fields directly instead of using GetXXX methods.